tests/models/test_data.py

```diff
@@ -72,41 +98,101 @@ def __init__(self):
                 self.data_files = []
                 self.normalizing_dict = norm_dict if normalize else None
                 self.to_tensor = None
-                self.surrounding_pixels = 1
+                self.experiment = experiment
+                self.surrounding_pixels = surrounding_pixels
 
         base_iterator = _BaseIter(MockLoader())
 
-        arrays = base_iterator.ds_folder_to_np(tmp_path, return_latlons=True,
+        arrays = base_iterator.ds_folder_to_np(data_dir, return_latlons=True,
                                                to_tensor=to_tensor)
 
-        x_np, y_np, latlons = arrays.x.historical, arrays.y, arrays.latlons
+        x_train_data, y_np, latlons = (
+            arrays.x, arrays.y, arrays.latlons
+        )
+        assert isinstance(x_train_data, TrainData)
+        if not to_tensor:
+            assert isinstance(y_np, np.ndarray)
+
+        expected_features = 4 if surrounding_pixels is None else 4 * 9
+        assert x_train_data.historical.shape[-1] == expected_features, "" \
+            "There should be" \
+            "4 historical variables (the final dimension):" \
+            f"{x_train_data.historical.shape}"
+
+        if experiment == 'nowcast':
+            expected_shape = (25, 3) if surrounding_pixels is None else (9, 3 * 9)
```

Comment:
@gabrieltseng is this returning the expected shape you are expecting?

9 latitude/longitudes and 27 features (3 x 9 latlons)?

Why has the number of `lat lons` changed in the first dimension? Is this the behaviour we expect from `surrounding pixels`?

I would have expected slightly different: `(25, 3*9)` because we have 25 latlons and then 9 surrounding pixels each with 3 features ?

Reply:
the `latlon` array only returns the latitude and longitude of the target pixel. 

The "spatial" data of the surrounding pixels is currently encoded in the variable names - this is probably not the best way to do it though